### PR TITLE
BIP-0158: add test cases for OP_RETURN with op codes, empty filter

### PR DESCRIPTION
**Pull request by Roasbeef**
In this commit, we add a new test case for a filter built from a block
that has a transaction with an OP_RETURN which isn't followed by only
push data items. The prior implementation for btcd (which was used to
generated these test vectors), had a stricter check which caused it to
add extra items to the filter. We also add a case of a block that has a
single coinbase transaction, with that transaction having only an
OP_RETURN output. As a result, that filter will be "empty", and is
signalled by by a single zero (0x00) byte.

In order to make building the code that makes the test vectors
reproducible, we've added go.mod and go.sum files as well.

### CODE REVIEW

**Comment by tuxcanfly** [comment]
related: #733

File: bip-0158.mediawiki [hunk 1]
```diff
@@ -281,10 +281,10 @@ following items for each transaction in a block:
 
 Any "nil" items MUST NOT be included into the final set of filter elements.
 
-We exclude all <code>OP_RETURN</code> outputs in order to allow filters to
-easily be committed to in the future via a soft-fork. A likely area for future
-commitments is an additional <code>OP_RETURN</code> output in the coinbase
-transaction similar to the current witness commitment
+We exclude all outputs taht start with <code>OP_RETURN</code> in order to allow
```
Comment (jimpo) [inline bip-0158.mediawiki]: typo: that
